devito/types/equation.py

```diff
@@ -98,7 +98,7 @@ def _apply_coeffs(cls, expr, coefficients):
         if not mapper:
             return expr
 
-        return expr.xreplace(mapper)
+        return expr.subs(mapper)
```

Comment:
@FabioLuporini this was the issue. If you had something like `Eq(u, u.dx2 + 2 * u.laplace, substitutions=...)` then `xreplace` only replaces the firs appearance so it was using custom coeffs for some and standard coeffs for other. Code looks much better (less. weights and derivatives) with the fix.